src/main/java/atdd/path/auth/SignInResolver.java

```diff
@@ -29,14 +30,8 @@ public boolean supportsParameter(MethodParameter parameter) {
     public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
                                   NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
         String email = (String) webRequest.getAttribute("loginUserEmail", SCOPE_REQUEST);
-        if (Strings.isBlank(email)) {
-            return new User();
-        }
-
-        try {
-            return userService.findUserByEmail(email);
-        } catch (Exception e) {
-            return new User();
-        }
+        Optional<User> user = userRepository.findByEmail(email);
+        return user.orElse(null);
```

Comment:
```suggestion
        Optional<User> user = userRepository.findByEmail(email);
        return user.orElseGet(() -> new User());
```

null을 리턴하기보다 빈객체 혹은  Optional을 활용하는 것은 어떨까요

https://www.slideshare.net/gyumee/null-142590829
http://www.mimul.com/pebble/default/2015/11/11/1447231720346.html